scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/SortedBucketSink.java

```diff
@@ -378,18 +366,23 @@ public void populateDisplayData(Builder builder) {
     }
   }
 
-  /**
-   * Patch of {@code SortValues}'s SortValuesDoFn which acts on elements already encoded as bytes
-   * and can avoid the additional ser/de operation per element.
-   */
-  private static class SortBytesDoFn<K>
-      extends DoFn<KV<K, Iterable<KV<byte[], byte[]>>>, KV<K, Iterable<KV<byte[], byte[]>>>> {
+  private static class SortShardDoFn<K1, K2, V>
+      extends DoFn<KV<BucketShardId, Iterable<V>>, KV<BucketShardId, Iterable<byte[]>>> {
     private final BufferedExternalSorter.Options sorterOptions;
+    private final BucketMetadata<K1, K2, V> bucketMetadata;
+    final Coder<V> valueCoder;
+    private final Comparator<byte[]> bytesComparator = UnsignedBytes.lexicographicalComparator();
```

Comment:
since we've already copied over the `org.apache.beam.sdk.extensions.sorter` package into Scio... what if we implemented a custom Comparator to do primary and secondary sort in one fell swoop (for the case where the Sink has a secondary key present; otherwise we could use this Comparator as-is)? something like [this](https://stackoverflow.com/a/4805676) but for byte[]?

Reply:
We could definitely do that, was thinking about it as follow up work.